TutorLizard.BusinessLogic/Data/DataAccess.cs

```diff
@@ -159,41 +164,82 @@ public ScheduleItemRequest CreateScheduleItemRequest (int scheduleItemId, int us
     #region GetID
     private int GetNewUserID()
     {
-        if (_userList.Count() > 0)
+        if (_userList.Count() +1 > 1)
             return _userList.Max(x => x.Id) + 1;
         else
-            return 0;
+            return 1;
     }
 
     private int GetNewScheduleItemID()
     {
-        if (_scheduleItemList.Count() > 0)
+        if (_scheduleItemList.Count() +1 > 1)
             return _scheduleItemList.Max(x => x.Id) + 1;
         else
-            return 0;
+            return 1;
     }
+
     private int GetNewAdID()
     {
-        if (_adList.Count() > 0)
+        if (_adList.Count() +1 > 1)
             return _adList.Max(x => x.Id) + 1;
         else
-            return 0;
+            return 1;
     }
 
     private int GetNewAdRequestID()
     {
-        if (_adRequestList.Count() > 0)
+        if (_adRequestList.Count() +1 > 1)
             return _adRequestList.Max(x => x.Id) + 1;
         else
-            return 0;
+            return 1;
     }
 
     private int GetNewScheduleItemRequestID()
     {
-        if (_scheduleItemRequestList.Count() > 0)
+        if (_scheduleItemRequestList.Count() +1 > 1)
             return _scheduleItemRequestList.Max(x => x.Id) + 1;
         else
-            return 0;
+            return 1;
+    }
+    #endregion
+
+    #region Login
+    private bool DoesTheUserIdExist(int id)
+    {
+        if (_userList.Any(x => x.Id == id))
+            return true;
+
+        return false;
+    }
+
+    private bool DoesTheUsernameMatchUserId(int id, string username)
+    {
+        User tempUser = _userList.Single(x => x.Id == id);
+
+        if (tempUser.Name.ToLower() == username.ToLower()) 
+            return true;
+
+        return false;
+    }
+    private User ReturnActiveUser(int id)
+    {
+        User tempUser = _userList.Single(x => x.Id == id);
+        return tempUser;
+    }
+    public (bool isCorrect, User activeUser) IsLoginDataCorrect (int id, string username)
+    {
+        bool condition1 = DoesTheUserIdExist(id);
+        bool condition2 = DoesTheUsernameMatchUserId(id, username); 
+        
+        if (condition1 == false || condition2 == false)
+            return (first: false, last: null);
+        
+        return (first: true, last: ReturnActiveUser(id));
+    }
+
+    public bool LookForUserName(string username)
```

Comment:
Name is a little confusing. Look for suggests to me that you return what you find, not info that it exists.
However, it works fine, so it could stay that way.